### PR TITLE
CLOUD-257 add continuous integration tests for pxc operator

### PR DESCRIPTION
**Pull request by hors**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -43,9 +179,21 @@ pipeline {
                     }
                 }
             }
+            withCredentials([string(credentialsId: 'GCP_PROJECT_ID', variable: 'GCP_PROJECT'), file(credentialsId: 'gcloud-key-file', variable: 'CLIENT_SECRET_FILE')]) {
+                sh '''
+                    source /home/ec2-user/google-cloud-sdk/path.bash.inc
```
Comment (delgod) [inline Jenkinsfile]:
```suggestion
                    source $HOME/google-cloud-sdk/path.bash.inc
```

File: Jenkinsfile [hunk 2]
```diff
@@ -1,17 +1,101 @@
+void CreateCluster() {
```
Comment (delgod) [inline Jenkinsfile]:
```suggestion
void CreateCluster(String CLUSTER_PREFIX) {
```

File: Jenkinsfile [hunk 3]
```diff
@@ -1,17 +1,101 @@
...
+            gcloud container clusters create --zone us-central1-a $CLUSTER_NAME-$CLUSTER_PREFIX --cluster-version 1.12 --machine-type n1-standard-4 --preemptible --num-nodes=3
+            kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user jenkins@$GCP_PROJECT.iam.gserviceaccount.com
+        '''
+   }
+}
+void pushArtifactFile(String FILE_NAME) {
+    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+        sh """
+            S3_PATH=s3://percona-jenkins-artifactory/\$JOB_NAME/\$(git rev-parse --short HEAD)
+            aws s3 ls \$S3_PATH/${FILE_NAME} || :
```
Comment (delgod) [inline Jenkinsfile]: which line install aws-cli?

File: Jenkinsfile [hunk 4]
```diff
@@ -24,6 +108,58 @@ pipeline {
                 archiveArtifacts 'results/docker/TAG'
             }
         }
+        stage('Run Tests') {
+            parallel {
+                stage('E2E Basic Tests') {
+                    environment {
+                        CLUSTER_PREFIX = 'basic'
+                        KUBECONFIG = "/tmp/${env.CLUSTER_NAME}-basic"
```
Comment (delgod) [inline Jenkinsfile]: env variables... can we pass both of them via argument of functions?

File: Jenkinsfile (hunk 3 above)
Comment (hors) [reply Jenkinsfile]: It is installed by default on this instance.

**Pull request merged**